File: BankKycSystemdotnet/BankKycSystem/Entities/OtpEntity.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BankKycSystem.Entities
{
    
    public class OtpEntity
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long? Id { get; set; }

        [Required] 
        [Column("email")]
        public string? Email { get; set; }

        [Column("otp")]
        public string? Otp { get; set; }

        [Column("generated_at")]
        public DateTime? GeneratedAt { get; set; }

        [Column("expiry_at")]
        public DateTime? ExpiryAt { get; set; }

        [Column("is_used")]
        public bool? IsUsed { get; set; } = false;
    }
}

File: bank-kyc-system/src/main/java/com/bank/project/entity/OtpEntity.java
package com.bank.project.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class OtpEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column
    private String email;
    @Column
    private String otp;
    @Column
    private LocalDateTime generatedAt;
    @Column
    private LocalDateTime expiryAt;
    @Column
    private boolean isUsed;
}
